pyk/src/pyk/kcfg/explore.py

```diff
@@ -218,19 +218,20 @@ def extend_cterm(
         cut_point_rules: Iterable[str] = (),
         terminal_rules: Iterable[str] = (),
         module_name: str | None = None,
-    ) -> KCFGExtendResult:
+    ) -> list[KCFGExtendResult]:
```

Comment:
To re-use the information from the backend about next states, `extend_cterm` now returns a list of states that carries info on whether the next node after a step is vacuous, stuck, branch, or non-det branch.